### DISCUSSION TITLE
Need help understanding how to make moving kinematic object cause other (dynamic) objects to gain kinetic energy in collisions

### DISCUSSION THREAD

**Discussion by tksuoran** [jrouwe/JoltPhysics#178] [Q&A] [answered]
I have a simple editor where I can move selected object. During this move operation, the rigid body is set to kinematic state.

![jolt_help_needed](https://user-images.githubusercontent.com/994606/171349654-1b0a126b-429d-49c7-841a-d71c13b2b05b.png)

In my bullet physics backend, when I move kinematic sphere so that it collides with other (dynamic) objects, the other objects gain kinetic energy - they get velocity. But in my jolt physics backend, the other (dynamic) objects only move just enough so that they avoid intersecting with the kinematic object, without gaining any kinetic energy/velocity.

I would like to know what I would need to do in order to make my jolt backend behave the same as bullet backend. Thank you!

**Comment by jrouwe** [comment, marked as answer]
How are you moving the object? Are you using BodyInterface::MoveKinematic(..., delta_time) followed by a PhysicsSystem::Update(delta_time, ...)? If so, I would expect the dynamic objects to get the proper velocity.

**Comment by tksuoran** [comment]
I was using `BodyInterface::SetPositionAndRotation()` and this is very likely the cause.

**Comment by jrouwe** [comment]
Yeah, if you use SetPositionAndRotation then you do not adjust the velocity of the object that you're moving. This indeed means that the only response you get is that the objects will depenetrate as the other objects will see an object with zero velocity 'appearing' inside them.
